server/routes/induction/common/wantToAddQualificationsController.ts

```diff
@@ -0,0 +1,74 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { Assessment } from 'viewModels'
+import type { WantToAddQualificationsForm } from 'inductionForms'
+import InductionController from './inductionController'
+import WantToAddQualificationsView from './wantToAddQualificationsView'
+import dateComparator from '../../dateComparator'
+import YesNoValue from '../../../enums/yesNoValue'
+
+/**
+ * Abstract controller class defining functionality common to both the Create and Update Induction journeys.
+ */
+export default abstract class WantToAddQualificationsController extends InductionController {
+  /**
+   * Returns the Want to Add Qualifications view; suitable for use by the Create and Update journeys.
+   */
+  getWantToAddQualificationsView: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonerSummary, prisonerFunctionalSkills } = req.session
+    const functionalSkills = {
+      ...prisonerFunctionalSkills,
+      assessments: mostRecentAssessments(prisonerFunctionalSkills.assessments || []),
+    }
+    const wantToAddQualificationsForm = req.session.wantToAddQualificationsForm || createWantToAddQualificationsForm()
+    req.session.wantToAddQualificationsForm = undefined
+
+    const view = new WantToAddQualificationsView(
+      prisonerSummary,
+      this.getBackLinkUrl(req),
+      this.getBackLinkAriaText(req),
+      wantToAddQualificationsForm,
+      functionalSkills,
+    )
+    return res.render('pages/induction/prePrisonEducation/wantToAddQualifications', { ...view.renderArgs })
+  }
+}
+
+const mostRecentAssessments = (allAssessments: Array<Assessment>): Array<Assessment> => {
```

Comment:
This is currently copied from `QualificationsListController`, so we should put this somewhere common, but it's not my focus right now.